### PR TITLE
XOP-724: Add pool join constraint for hosts with bonds/VLANs/tunnels

### PR DESCRIPTION
**Pull request by robhoes**
Signed-off-by: Rob Hoes <rob.hoes@citrix.com>

### CODE REVIEW

**Comment by robhoes** [comment]
@stephen-turner, @mihaelastoica: Please note this new pool join constraint for XC.

File: ocaml/xapi/xapi_pool.ml [hunk 1]
```diff
@@ -190,21 +190,13 @@ let pre_join_checks ~__context ~rpc ~session_id ~force =
...
-		if not (my_network_bonds = []) then begin
-                        error "The current host has a network bond: it cannot join a new pool";
-                        raise (Api_errors.Server_error(Api_errors.pool_joining_host_cannot_contain_network_bond, []))
-                end in
-
-	let assert_management_interface_is_physical () =
-		let pifs = Db.PIF.get_refs_where ~__context ~expr:(And (
-			Eq (Field "management", Literal "true"),
+	let assert_only_physical_pifs () =
+		let pifs = Db.PIF.get_refs_where ~__context ~expr:(
```
Comment (johnelse) [inline ocaml/xapi/xapi_pool.ml]: Something like `let non_physical_pifs =` would be clearer.

File: ocaml/xapi/xapi_pool.ml [hunk 2]
```diff
@@ -190,21 +190,13 @@ let pre_join_checks ~__context ~rpc ~session_id ~force =
...
 			Eq (Field "physical", Literal "false")
-		)) in
+		) in
 		if pifs <> [] then begin
-			error "The current host has a management interface which is not physical: cannot join a new pool";
-			raise (Api_errors.Server_error(Api_errors.pool_joining_host_must_have_physical_management_nic, []));
+			error "The current host has a network bonds, VLANs or tunnels: it cannot join a new pool";
```
Comment (johnelse) [inline ocaml/xapi/xapi_pool.ml]: Typo in the log line - the first "a" should be removed.

**Comment by johnelse** [comment]
Couple of minor comments, but the logic looks fine.

**Comment by robhoes** [comment]
@johnelse: Thanks, I've updated the patch.

**Comment by johnelse** [comment]
Looks good! :shipit: :shipit:

**Comment by robhoes** [comment]
Double squirrel!

**Pull request merged**